docs/ai/how-to/app-service-entra-auth.md

```diff
@@ -0,0 +1,155 @@
+
+:::zone-end
+
+:::zone target="docs" pivot="azure-cli"
+
+1. Create a user-assigned identity:
+
+    ```azurecli
+    az identity create --resource-group <group-name> --name <identity-name>
+    ```
+
+1. Assign the identity to your app:
+
+    ```azurecli
+    az webapp identity assign --resource-group <group-name> --name <app-name> --identities <identity-id>
+    ```
+
+:::zone-end
+
+## Add an Azure OpenAI user role to your managed identity
+
+:::zone target="docs" pivot="azure-portal"
+
+1. In the [Azure Portal](https://aka.ms/azureportal), navigate to the scope you want to grant **Azure OpenAI** access to.
+    * This can be a **Management group**, **Subscription**, **Resource group**, or a specific **Azure OpenAI** resource.
+1. Select **Access control (IAM)** on the left navigation pane.
+1. Select **Add**, then select **Add role assignment**.
+1. On the **Role** tab, select the **Cognitive Services OpenAI User** role.
+1. On the **Members** tab, select the managed identity.
+1. On the **Review + assign** tab, select **Review + assign** to assign the role.
+
+:::zone-end
+
+:::zone target="docs" pivot="azure-cli"
+
+**Resource scope**
+
+```azurecli
+az role assignment create --assignee "{managedIdentityObjectID}" \
+--role "Cognitive Services OpenAI User" \
+--scope "/subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}/providers/{providerName}/{resourceType}/{resourceSubType}/{resourceName}"
+```
+
+**Resource group scope**
+
+```azurecli
+az role assignment create --assignee "{managedIdentityObjectID}" \
+--role "Cognitive Services OpenAI User" \
+--scope "/subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}"
+```
+
+**Subscription scope**
+
+```azurecli
+az role assignment create --assignee "{managedIdentityObjectID}" \
+--role "Cognitive Services OpenAI User" \
+--scope "/subscriptions/{subscriptionId}"
+```
+
+**Management group scope**
+
+```azurecli
+az role assignment create --assignee "{managedIdentityObjectID}" \
+--role "Cognitive Services OpenAI User" \
+--scope "/providers/Microsoft.Management/managementGroups/{managementGroupName}"
+```
+
+:::zone-end
+
+## Implement token-based authentication with the Semantic Kernel SDK
+
+1. Initialize a `DefaultAzureCredential` object to pickup your app's manged identity:
+
+    :::code language="csharp" source="./snippets/semantic-kernel/IdentityExamples.cs" id="tokenCredential":::
+
+1. Build a `Kernel` object including the Azure OpenAI Chat Completion Service, use the perviously created credentials:
+
+    :::code language="csharp" source="./snippets/semantic-kernel/IdentityExamples.cs" id="kernelBuild":::
+
+1. Use the `Kernel` object to invoke prompt completion through Azure OpenAI:
+
+    :::code language="csharp" source="./snippets/semantic-kernel/IdentityExamples.cs" id="invokePrompt":::
+
+## Related content
+
+<!-- Update links once the other docs are done -->
+
+* [Data Security with App Service]<!-- (link.md) -->
```

Comment:
Add link here? Or comment out for now?

Reply:
This doc is coming next, so no link for now